ibc-clients/ics07-tendermint/src/client_state/misbehaviour.rs

```diff
@@ -116,12 +117,14 @@ where
     // main header verification, delegated to the tendermint-light-client crate.
     let untrusted_state = header.as_untrusted_block_state();
 
-    let tm_chain_id = &chain_id
-        .as_str()
-        .try_into()
-        .map_err(|e| ClientError::Other {
-            description: format!("failed to parse chain id: {e}"),
-        })?;
+    let tm_chain_id =
+        &chain_id
+            .as_str()
+            .try_into()
+            .map_err(|e| IdentifierError::FailedToParse {
```

Comment:
I'm thinking we align the fields under `IdentifierError::FailedToParse` with our convention and use a single `description` field. However, we can still include values, like in this case, as part of that description.